### ISSUE TITLE
A cube now needs a dcterms:creator for Visualize, but this is undocumented, unchecked, etc...

### ISSUE THREAD

**Issue by FabianCretton** [zazuko/cube-creator#1275]
I discovered a few days ago that some of my cubes no more appear amongst Visualize datasets, and the answer from IXT was that the cube needs a dcterms:creator.

Today I did create a new cube for some tests, and it did not appear either on Visualize.
After some thoughts, I found out that I needed to add an Organisation on the tab open-data.swiss, in the cube's metadata edition of the Cube Designer.

All this needs IMHO:
- to be clarified: is it correct that Visualize now requires a dcterms:creator, and that the way to add it is the Organisation of the  open-data.swiss tab
- to be improved: if this is finally correct, it should not be possible to publish a cube for visualize but without an Organisation
- and more generally, we need a way to synchronize all those decisions: when a triple becomes necessary for Visualize, as an example, that this must become obvious in the Cube Creator

Thank you for your feedback

**Comment by l00mi** [comment]
The needed properties are documented here: https://visualize.admin.ch/docs/charts/rdf-to-visualize. If there is a need for more details please contact @ptbrowne. 

It is a sensible idea to restrict publication in case the target visualize.admin.ch is selected with some constraints. That would need a new issue.

**Comment by ptbrowne** [comment]
@FabianCretton On our side we have prepared a tool to tell you when something is wrong from the point of view of visualize https://test.visualize.admin.ch/_cube-checker. It is linked from the docs that Michael linked to you in the post above but for now it has not been published on prod yet, only on test.

**Comment by FabianCretton** [comment]
@l00mi 
For further discussion, but to take action (adaptation needed IMHO on the Cube Creator's GUI):  
**From the Cube Creator perspective:**  
How do we tell the user, that when he wants to publish a cube to Visualize, he must go to the "Opendata.swiss" tab and enter an Organization ?
**From Visualize perspective:**  
How do we know, when [Visualize says](https://visualize.admin.ch/docs/charts/rdf-to-visualize) that a triple "?cube dcterms:creator ?value" is needed, that he must go back to the Cube Designer, to the "Opendata.swiss" tab and enter an Organization ?

**Todo/Proposal:**
Should there be a new tab for fields that are mandatory according to the "Publish To" options that is choosen (visualize and/or opendata.swiss currently)

And I come back to the "developper" point of view, to handle this kind of case (and this is also the end-user concern when trying to understand why something do not work):  
"we/you need a way to synchronize all those decisions: when a triple becomes necessary for Visualize, as an example, that this must become obvious in the Cube Creator"

What do you think ?

**Comment by FabianCretton** [comment]
About how to improve the Cube Creator/Visualize integration, I can also relate to this discussion where a "similar" problem is occuring: https://zulip.zazuko.com/#narrow/stream/40-bafu-ext/topic/Hierarchie.20removed.20from.20cube/near/283252

**Comment by l00mi** [comment]
That sounds like a good plan, we would need an epic issue for this to further go into this.

**Comment by FabianCretton** [comment]
Hi @ptbrowne and @l00mi, i come back to you both with my remaining question that, I think, we need to clarify and do something about it (simple fixes could take 15 minutes not more):
How does a Cube Creator user know which fields he **must** input when he chooses "publish to visualize" ?

The problem: 
I am teaching the Cube Creator these days, and saying "oh yes, by the way, if you choose "publish to visualize", do not forget to enter an Organisaton on the "opendata.swiss" tab. Please take a note....

You gave me two informations:
-  The RDF needed by visualize is described here: https://visualize.admin.ch/docs/charts/rdf-to-visualize
However: there is no way to know that the required dcterms:creator is filled on the "Opendata.swiss" tab - Organization field
- There is a tool, for visualize, to check the cubes: https://test.visualize.admin.ch/_cube-checker
However: it should be made clearer, BEFORE publishing a cube, what is needed when "publish to visualize" is choosen.

I see different options for the end-user about this, which need input from you and others of course:
- reorganizing the fields on the Cube Designer GUI, as it makes no sense to have the Organisation on the "opendata.swiss" tab. Could this field be on the main "Cube metadata" tab ?
- should we describe in the Help pages those kind of things ? There are already mentions of Visualize here: https://github.com/zazuko/cube-creator/wiki/3.-Cube-Designer, for instance when explaining the Scale of Measure.

Improving the current situation is certainly a matter of minutes:
- maybe change the tab of the Organisation field if it makes sens
- add an information in the User Guide (I can do it)

But this should also be tackled in a generic approach, to maintain Cube Creator and Visualize synchronized, for instance if in 1 month a new field becomes mandatory for a cube on Visualize.

Thank you for your help
Fabian

**Comment by l00mi** [comment]
Hi Fabian,

We can definitely do UX optimisations on the Cube-Creator side. We already did this once some months back, when we went through each panel and decided on rearranging. But I would wait currently from this side a bit, because of two reasons:
* I think now after the experience collected, we will need to adapt some of the metadata. (E.g. I think we should add two new fields for the cubes: 1) For abrev. like "WASTA" to not have this somewhere just in the title, once in the beginning once at the end, once with parantheses, once without. 2) For subtitles, which can be used, if the same dataset is provided with multiple aspects.. ). All this needs to be scheduled in an Issue and discussed.
* Also I am not sure about the current minimal data need from visualize.admin.ch, I would prefer to let this settle first, in particular the full text search part. Because else we will adapt on both sides in rounds. It might also be possible that the need for the dcterms:creator can be removed in visualize? What is the consensus here from BAFU site?

**Comment by FabianCretton** [comment]
Thank you @l00mi Michael.

@ptbrowne What is the process on IXT side, by which you decide which field becomes mandatory for a cube ?
I mean, during the summer, it seems that the dcterms:creator became mandatory (see this discussion/issue here above).
I thought it was decided with Zazuko, but it seems not.
It is thus important to notice that when you decide something, it should be talked about with the different parties (certainly BAFU, maybe Zazuko, maybe myself, etc.), to see how to synchronize the 2 tools (Cube Creator and Visualize).

A very practical example that happened yesterday with this dcterms:creator: Tobias and Noël who now work with the Cube Creator for the BAFU did publish test cubes that they could not find in Visualize. The reason was that dcterms:creator was missing...of course...and eventhough I did tell them aboutt this. But now they know.
So @ptbrowne, to better understand the "problem": now that dcterms:creator is mandatory for a cube, the user of the Cube Creator need to go to the "Opendata.swiss" tab and enter an Organisation :
![image](https://user-images.githubusercontent.com/6167669/189050017-2fde7b12-31e7-4c19-9c88-555aca1be8d2.png)

As you see, your decision of making dcterms:creator mandatory has an impact on the end-user and the Cube Creator interface.
Not to mention: I had quiet some job to find out that the dcterms:creator was filled by the Organisation field of the UI.

For the documentation point of View, visualize has [a list of needed RDF properties](https://visualize.admin.ch/docs/charts/rdf-to-visualize), would there be a way to explain somewhere the link between those properties and the Cube Creator GUI (I am aware that a cube could come from somewhere else than the Cube Creator). @l00mi any idea ?  

I just think that we need to clarify the links between the two tools: in the GUI, in the documentation.

**Comment by FabianCretton** [comment]
@l00mi Michael @ptbrowne Today I will update the Cube Designer page so that this information about "Organisation" is not only in my head, but at disposal for the users.

@ptbrowne I will mention the cube checker is it ok for you ? 
If so, what URL should I provide ? I guess that [visualize.admin.ch](https://visualize.admin.ch/_cube-checker?dataSource=Prod) can only check cubes that are in PROD (no Data Source choice), so I might mention instead [test.visualize](https://test.visualize.admin.ch/en/_cube-checker?dataSource=Test)

**Comment by ptbrowne** [comment]
> @ptbrowne What is the process on IXT side, by which you decide which field becomes mandatory for a cube ?

We try not to make things mandatory, I think the last time it happened, it was because of improvements on the search, and some fields were made required when in fact they should still be optional (so it was a bug on our part). In case we require a field, we will for sure coordinate with you before hand.

Yes it is OK for me, you can mention [int.visualize.admin.ch](https://int.visualize.admin.ch/en/_cube-checker?dataSource=Int).

**Comment by l00mi** [comment]
@FabianCretton yes, please go ahead. Thank you! With this I will close the issue with this for now.

(Except the word _disposal_ was probably not the one you were looking for in your comment above (-: )
